### PR TITLE
Changed EscapeRegex method's location, added a test and commented both.

### PR DESCRIPTION
**Pull request by Exilon24**
## Fixed [Issue 338](https://github.com/featbit/featbit/issues/338)

* **Created** a `StringHelper` class (If one existed, I couldn't find it).
* **Moved** `EscapeRegex()` method from the `ResourceHelper` class to the `StringHelper` class.
*  **Changed** the regex expression to `([.*+?^=!:${}()|[\]/\\])` and the `\$1` pattern.
* **Added** Comments.
* **Included** new package: `MSTest.TestFramework` to `Domain` for unit testing.
* **Added** a unit test for `EscapeRegex()`.

Tell me if there are any issues and I'll try fix them. I have little to no experience with regex :)

### CODE REVIEW

**Comment by deleteLater** [comment]
Is this pr ready for review :thinking:

**Comment by Exilon24** [comment]
Yes :)

I just had to do something.

**Comment by deleteLater** [comment]
Hi @Exilon24, Thanks for your work on this PR :thumbsup: 

I've reviewed your code and have a few feedbacks for you to fix first:
- The `.vs` folder should not be committed to source control
- Move `StringHelper` to `/src/Domain/Utils` folder
- Create a folder called `Utils` in the `/tests/Domain.UnitTests` project. Then, create a file called `StringHelperTests` in that folder and then move your unit tests there.
- All our test project has integrated [xunit](https://xunit.net/) as our unit test tool, so you don't need to add `MSTest.TestFramework` anymore.

**Comment by deleteLater** [comment]
Another suggestion is to write parameterized tests for `EscapeRegex` method. Here is an example test using [InlineData](https://andrewlock.net/creating-parameterised-tests-in-xunit-with-inlinedata-classdata-and-memberdata/#using-the-theory-attribute-to-create-parameterised-tests-with-inlinedata-). Or you can use [TheoryData](https://andrewlock.net/creating-strongly-typed-xunit-theory-test-data-with-theorydata/#using-theorydata-with-the-classdata-attribute) if you like.

```csharp
[Theory]
[InlineData("hello.world", @"hello\.world")]
[InlineData("hello*world", @"hello\*world")]
// ...and more test cases
public void EscapeRegexTests(string input, string expected)
{
    var actual = StringHelper.EscapeRegex(input);

    Assert.Equal(expected, actual);
}
```

**Comment by Exilon24** [comment]
@deleteLater Thank you for the feedback! I've gone over and nearly done everything that you've told me to do except for one thing. I can't find a way to not commit the .vs folder...
image

**Comment by deleteLater** [comment]
It seems like our `.gitignore` file not working in your Visual Studio.

https://github.com/featbit/featbit/blob/a37477e6430474fcd3d9b88687f3fdb7c80769c8/modules/back-end/.gitignore#L35-L36

Maybe you can try the solution here: https://stackoverflow.com/questions/51077976/gitignore-for-visual-studio-project-is-not-working

If it doesn't work, you can delete .vs folder, then push the change.

**Comment by Exilon24** [comment]
I managed to fix it! I had to include `.vs/` to the main `.gitignore` of the project. I 've also added some extra test cases :)

Just tell me if anything needs changing 👍

File: modules/back-end/src/Domain/Domain.csproj [hunk 1]
```diff
@@ -8,4 +8,8 @@
         <NoWarn>$(NoWarn);1591</NoWarn>
     </PropertyGroup>
 
+    <ItemGroup>
+      <PackageReference Include="MSTest.TestFramework" Version="3.0.2" />
+    </ItemGroup>
+
```
Comment (deleteLater) [inline modules/back-end/src/Domain/Domain.csproj]: revert this change

File: .vs/VSWorkspaceState.json
Comment (deleteLater) [inline .vs/VSWorkspaceState.json:1]: This file should not be committed.

File: .vs/featbit/v17/.wsuo
Comment (deleteLater) [inline .vs/featbit/v17/.wsuo:1]: This file should not be committed.

File: modules/back-end/tests/Domain.UnitTests/Utils/StringHelperTests.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+﻿using Domain.Utils;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Domain.UnitTests.Utils
+{
```
Comment (deleteLater) [inline modules/back-end/tests/Domain.UnitTests/Utils/StringHelperTests.cs]: prefer use file scoped namespace here

```csharp
namespace Domain.UnitTests.Utils;

public class StringHelperTests
{
    // ...
}
```

File: modules/back-end/src/Domain/Utils/StringHelper.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Text.RegularExpressions;
+
+namespace Domain.Utils;
+public static class StringHelper
```
Comment (deleteLater) [inline modules/back-end/src/Domain/Utils/StringHelper.cs]: nit: add space between namespace and class declaration

```csharp
namespace Domain.Utils;

public static class StringHelper
{
    // ...
}
```

File: modules/back-end/src/Domain/Utils/StringHelper.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
```
Comment (deleteLater) [inline modules/back-end/src/Domain/Utils/StringHelper.cs]: remove this unused using

**Comment by deleteLater** [comment]
@Exilon24 I've added some comments

**Comment by Exilon24** [comment]
I've resolved all of them. I assume that actually makes the changes.

**Comment by deleteLater** [comment]
> I've resolved all of them. I assume that actually makes the changes.

I didn't see any changes here: https://github.com/featbit/featbit/pull/339/files

 Did you push those changes :thinking: ?

**Comment by Exilon24** [comment]
Should all be fixed now :)

Again, tell me if I need to change anything.

File: modules/back-end/src/Domain/Utils/StringHelper.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace Domain.Utils;
+
+public static class StringHelper
+{
+    /// <summary>
+    /// This regular expression matches any single character that is one of the following special characters:
+    /// . * + ? ^ = ! : $ { } ( ) | [ ] / \
+    /// It can be used to search for any of these special characters in a given text or string.
+    /// </summary>
```
Comment (deleteLater) [inline modules/back-end/src/Domain/Utils/StringHelper.cs]: I think this summary is not clear enough, how about

```csharp
/// <summary>
/// Escapes a set of characters (. * + ? ^ = ! : $ { } ( ) | [ ] / \) by replacing them with their escape codes.
/// This converts a string so that it can be used as a constant within a regular expression safely.
/// </summary>
```

Comment (Exilon24) [reply modules/back-end/src/Domain/Utils/StringHelper.cs]: Sure. Thats the best ChatGPT could give me lol.

**Review by deleteLater — APPROVED** [review]
LGTM! Thanks for your time and effort :handshake: @Exilon24

**Comment by cosmos-explorer** [comment]
@all-contributors
please add @Exilon24 for code
